File: Practica 8/Problema2/Kotlin/Fish.kt
import AbstractAnimal

class Fish(name: String, year: Int) : AbstractAnimal(name, year) {
    override fun move(): String {
        return "swim"
    }

    override fun breathe(): String {
        return "gills"
    }

    override fun reproduce(): String {
        return "eggs"
    }

    override fun getName(): String {
        return super.getName()
    }
}

File: Practica 8/Problema2/Dart/Fish.dart
import 'AbstractAnimal.dart';

class Fish extends AbstractAnimal {
  Fish(String name, int year) : super.namedYear(name, year);

  @override
  String move() {
    return "swim";
  }

  @override
  String breathe() {
    return "gills";
  }

  @override
  String reproduce() {
    return "eggs";
  }
}
